src/OSPSuite.Core/Domain/Services/ConcurrencyManager.cs

```diff
@@ -82,26 +84,25 @@ public class ConcurrencyManager : IConcurrencyManager
                var result = await returnWithExceptionHandling(
                   coreIndex,
                   cancellationToken,
-                  action, 
+                  action,
                   datum,
                   id
                );
                results.TryAdd(datum, result);
             }
-         }).ToList();
+         });
```

Comment:
@abdelr Why do we need the ToList here?

Reply:
Maybe left behind when turning from synchronous to async